### PR TITLE
Fixes for odoc

### LINKED ISSUE #1093: Switch to odoc?
Instead of ocamldoc.

Should we?

Can this be done automatically? I.e. update all the ocamldoc comments, if needed.

### PR DESCRIPTION
**Pull request by sim642**
Closes #1093.

`make doc` already uses odoc and it builds, but with many warnings. This PR fixes a bunch of the fixable ones (there are still all the "unresolvedroot" ones which I'm not sure how should be fixed):
1. Fix heading levels to start from 1, not 6.
2. Fix module synopses by separating them with empty lines.
3. Fix `@since` and `@raise` tags.
4. Replace `@added` and `@obsolete` tags with standard `@since` and `@deprecated`.
5. Fix some ambiguous references.

### CODE REVIEW

File: src/batChar.mli [hunk 1]
```diff
@@ -81,14 +81,14 @@ val is_lowercase : char -> bool
 val is_uppercase_latin1: char -> bool
 (** Determine if a character is uppercase Latin 1.
     A character is uppercase Latin 1 if it is between
-    ['A'] and ['Z'], between ['�'] and ['�'] or
-    between ['�'] and ['�'] *)
+    ['A'] and ['Z'], between ['�'] and ['�'] or
+    between ['�'] and ['�'] *)
```
Comment (sim642) [inline src/batChar.mli:85]: Looks like VSCode messed something up here?

**Comment by UnixJunkie** [comment]
I'll have a look, thanks.

**Comment by UnixJunkie** [comment]
Since it is a large PR, it might take me some time though.
I'll try to do this before the end of the day.

**Comment by UnixJunkie** [comment]
Looks good to me.
Thanks for taking care of batteries.
This must have been crazily annoying to fix so many odoc warnings!

**Pull request merged**